File: 1344.时钟指针的夹角.go
/*
 * @lc app=leetcode.cn id=1344 lang=golang
 *
 * [1344] 时钟指针的夹角
 */

// @lc code=start
func angleClock(hour int, minutes int) float64 {
	hour_angle := float64(hour%12)*30 + float64(minutes)/2
	minute_angle := float64(minutes) * 6
	angle := math.Abs(hour_angle - minute_angle)
	return math.Min(angle, 360-angle)

}

// @lc code=end

File: 1344.时钟指针的夹角.py
#
# @lc app=leetcode.cn id=1344 lang=python3
#
# [1344] 时钟指针的夹角
#

# @lc code=start
class Solution:
    def angleClock(self, hour: int, minutes: int) -> float:
        hour_angle = (hour % 12) * 30 + minutes / 2 # 时针每小时转30度，每分钟转0.5度
        minute_angle = minutes * 6
        angle = abs(hour_angle - minute_angle)
        return min(angle, 360 - angle)  
    

s = Solution()
print(s.angleClock(12, 30))
# @lc code=end

